app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEPresenter.java

```diff
@@ -994,13 +1000,24 @@ public void setProgramForTesting(Program program) {
         selectedProgram = program;
     }
 
+    @Override
+    public void clearOtherFiltersIfWebAppIsConfig() {
+        List<FilterItem> filters = filterRepository.homeFilters();
+        disableHomeFilters.execute(filters);
+    }
+
     @Override
     public void checkFilters(boolean listResultIsOk) {
+        boolean hasToShowFilters = !filterRepository.
+                programFilters(currentProgram.blockingFirst()).isEmpty();
+
         if (listResultIsOk) {
-            view.setFiltersVisibility(true);
-        } else {
+            view.setFiltersVisibility(hasToShowFilters);
+        } else if (!listResultIsOk && hasToShowFilters){
             boolean filtersActive = FilterManager.getInstance().getTotalFilters() != 0;
             view.setFiltersVisibility(filtersActive);
+        } else if (!listResultIsOk && !hasToShowFilters){
+            view.setFiltersVisibility(false);
```

Comment:
@Balcan I need your knowledge here :D . I think is okay but just to make sure...

Reply:
It is ok